### PR TITLE
moving stitch into Master

### PR DESCRIPTION
**Pull request by jeinglis**

### CODE REVIEW

File: src/store/modules/activeRows/actions.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+const activeRowsActions = {
+  setActiveFarmer: (currentFarmerUUID: string): Action => ({
+    currentFarmerUUID,
+    type: 'UPDATE_CURRENT_FARMER',
+  }),
+  clearActiveFarmer: (): Action => ({
+    type: 'CLEAR_CURRENT_FARMER',
+  }),
+};
```
Comment (mfalthaw) [inline src/store/modules/activeRows/actions.ts]: I think we should add actions for MilkEntry here

File: src/store/modules/activeRows/types.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+/*----------------------- State -----------------------*/
+/** currentFarmer State*/
+export interface ActiveRowsState {
+  activeFarmerUUID: string;
+}
+
+/** currentMilkEntry State*/
+export interface ActiveRowsState {
+  activeMilkEntryUUID: string;
+}
```
Comment (jeinglis) [inline src/store/modules/activeRows/types.ts]: combine these

File: src/store/modules/activeRows/reducer.ts [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
       return {
         ...state,
-        currentFarmerUUID : '',
+        activeFarmerUUID : '',
       };
     }
     default:
       return state;
   }
 };
```
Comment (mfalthaw) [inline src/store/modules/activeRows/reducer.ts]: I think we should add MilkEntry stuff here too

File: src/view/components/AddEntry/index.tsx [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+      costPerUnit: this.state.costPerUnit,
+      currency: 'UGX',
+      quality: this.state.quality,
+    };
+    this.props.createMilkEntry(newEntry);
+    this.props.navigate(Route.FARMER);
+  }
+
+  /**
+   * Returns a button with text specified
```
Comment (mfalthaw) [inline src/view/components/AddEntry/index.tsx]: We should have more docs here

File: src/view/components/EditEntry/index.tsx [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+import { getActiveMilkEntry } from '../../../store/modules/milk/selectors';
+
+
+interface OwnPropsType {
+}
+
+interface DispatchPropsType {
+  navigate(route: Route): void;
+  goBack(): void;
+  updateDairy(newEntry: ThunkUpdateRow<MilkEntry>): void;
```
Comment (mfalthaw) [inline src/view/components/EditEntry/index.tsx]: This should be `MilkEntry` instead.

File: src/view/components/AddEntry/index.tsx [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+/**
+ * Button color
+ */
+type ButtonColor = 'PRIMARY' | 'INFO';
+
+/**
+ * AddEntry page
+ * @example 
+ *             <AddEntry
+ *             />
```
Comment (jeinglis) [inline src/view/components/AddEntry/index.tsx]: is this a left over?

File: src/view/navigation/index.tsx [hunk 7]
```diff
@@ -15,13 +15,17 @@ import drawerActions from '../../store/modules/drawer/actions';
...
+import { getActiveFarmer } from '../../store/modules/farmer/selectors'; 
 
 interface StorePropsType {
   nav: NavigationState;
   headerShown: boolean;
+  routeName: string;
   routeType: PageType | undefined;
   searchBarShown: boolean;
   searchPlaceholder: string | undefined;
+  currentFarmer: Farmer;
```
Comment (mfalthaw) [inline src/view/navigation/index.tsx]: Should we add `currentMilkEntry` here too? to update the header of the page to the entry we clicked on

**Pull request merged**